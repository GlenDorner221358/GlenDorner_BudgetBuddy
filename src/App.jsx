import Col from 'react-bootstrap/Col'
import Container from 'react-bootstrap/Container'
import Row from 'react-bootstrap/Row'
import './App.css'
import Income from './components/Income'
import Expenses from './components/Expenses'
import Taxes from './components/Taxes'
import TotalCard from './components/items/TotalCard'
import { dummyCards, lastTotalCard } from './utils'
import Savings from './components/Savings'
import LastTotalCard from './components/items/LastTotalCard'
import { Button } from 'react-bootstrap'

function App() {

  return (
    <Container fluid>
      <Row >
        <Col xs={12} className='topbar p-3 mb-4'>
          <h2 className='title'>
            <img src='logo512.png' width={60} height={60} />
            <span>BudgetBuddy</span>
          </h2>
          <Button><img src='reset.png' width={30} height={30} /></Button>
        </Col>
      </Row>
      <Row>
        <Col xs={12} sm={12} md={5}>
          <div className='card shadow-sm p-3 mb-4 bg-white rounded'>
            <Income />
          </div>
          <div className='card shadow-sm p-3 mb-4 bg-white rounded'>
            <Taxes />
          </div>
          <div className='card shadow-sm p-3 mb-4 bg-white rounded'>
            <Savings />
          </div>
        </Col>
        <Col xs={12} md={4} >
          <div className='card shadow-sm p-3 mb-4 bg-white rounded'>
            <Expenses />
          </div>
        </Col>

        <Col xs={12} md={3}>
          {dummyCards.map((total, index) => (
              <div key={index} className='card shadow-sm p-3 mb-4 rounded'>
                <TotalCard total={total} />
              </div>
          ))}
            <div className='card shadow-sm p-3 mb-4 rounded last-total-card'>
              <LastTotalCard total={lastTotalCard} />
            </div>
        </Col>
      </Row>
    </Container>
  )
}

export default App
