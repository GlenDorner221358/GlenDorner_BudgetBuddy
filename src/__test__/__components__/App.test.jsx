import { render, screen } from "@testing-library/react"
import App from "../../App"

// Here we test if the app renders or not

test("Create a snapshot of the app when rendered for first time", () => {
    render(<App />)

    var app = screen.getByTestId("outer-app")

    expect(app).toBeInTheDocument()
    
    // take a snapshot
    expect(app).toMatchSnapshot()
})
