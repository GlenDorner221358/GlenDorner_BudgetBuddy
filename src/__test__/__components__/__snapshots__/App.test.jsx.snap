// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create a snapshot of the app when rendered for first time 1`] = `
<div
  class="container-fluid"
  data-testid="outer-app"
>
  <div
    class="row"
  >
    <div
      class="topbar p-3 mb-4 col-12"
    >
      <h2
        class="title"
      >
        <img
          height="60"
          src="logo512.png"
          width="60"
        />
        <span>
          BudgetBuddy
        </span>
      </h2>
      <button
        class="btn btn-primary"
        type="button"
      >
        <img
          height="30"
          src="reset.png"
          width="30"
        />
      </button>
    </div>
  </div>
  <div
    class="row"
  >
    <div
      class="col-md-5 col-sm-12 col-12"
    >
      <div
        class="card shadow-sm p-3 mb-4 bg-white rounded"
      >
        <div>
          <h3>
            Household Montly Salaries
          </h3>
          <div
            class="form-row"
          >
            <select
              autocomplete="off"
              class="form-select"
              data-testid="iconInput"
              name="icon"
            >
              <option
                disabled=""
                selected=""
              >
                -
              </option>
              <option
                value="üë©"
              >
                üë©
              </option>
              <option
                value="üë±‚Äç‚ôÇÔ∏è"
              >
                üë±‚Äç‚ôÇÔ∏è
              </option>
              <option
                value="üê∂"
              >
                üê∂
              </option>
              <option
                value="üê±"
              >
                üê±
              </option>
              <option
                value="üê∞"
              >
                üê∞
              </option>
              <option
                value="üêª"
              >
                üêª
              </option>
              <option
                value="ü¶Å"
              >
                ü¶Å
              </option>
              <option
                value="üê∑"
              >
                üê∑
              </option>
              <option
                value="üêÆ"
              >
                üêÆ
              </option>
            </select>
            <input
              autocomplete="off"
              class="form-control"
              data-testid="nameInput"
              id="name"
              name="name"
              placeholder="Member Name"
              type="text"
              value=""
            />
            <input
              autocomplete="off"
              class="form-control"
              data-testid="incomeInput"
              id="income"
              name="income"
              placeholder="0.00"
              step="0.01"
              type="number"
              value="0"
            />
            <button
              class="add-income btn btn-primary"
              disabled=""
              role="button"
              type="button"
            >
              Add
            </button>
          </div>
        </div>
      </div>
      <div
        class="card shadow-sm p-3 mb-4 bg-white rounded"
      >
        <div
          data-toggle="tooltip"
          title="Those bastards!"
        >
          <h3>
            Tax Bracket Calculation
          </h3>
          <div
            class="scroll-row hide-scroll"
          />
        </div>
      </div>
      <div
        class="card shadow-sm p-3 mb-4 bg-white rounded"
      >
        <div>
          <div
            class="title-row"
          >
            <h3>
              Savings Calculation
            </h3>
            <span>
              <p>
                % you want to save
              </p>
              <select
                autocomplete="off"
                class="form-select"
                id="percentage"
                name="percentage"
              >
                <option
                  disabled=""
                  value="0"
                >
                  -
                </option>
                <option
                  value="5"
                >
                  5
                  %
                </option>
                <option
                  value="10"
                >
                  10
                  %
                </option>
                <option
                  value="15"
                >
                  15
                  %
                </option>
                <option
                  value="20"
                >
                  20
                  %
                </option>
              </select>
            </span>
          </div>
          <div
            class="scroll-row hide-scroll"
          />
        </div>
      </div>
    </div>
    <div
      class="col-md-4 col-12"
    >
      <div
        class="card shadow-sm p-3 mb-4 bg-white rounded"
      >
        <div>
          <h3>
            Household Monthly Expenses
          </h3>
          <div
            class="form-row"
          >
            <input
              autocomplete="off"
              class="form-control"
              data-testid="expenseName"
              id="title"
              name="title"
              placeholder="Expense Title"
              type="text"
              value=""
            />
            <input
              autocomplete="off"
              class="form-control"
              data-testid="expenseCost"
              id="expense"
              name="expense"
              placeholder="0.00"
              step="0.01"
              type="number"
              value="0"
            />
            <button
              class="add-expenses btn btn-primary"
              disabled=""
              role="button"
              type="button"
            >
              Add
            </button>
          </div>
          <div
            class="expense-outer hide-scroll"
          />
        </div>
      </div>
    </div>
    <div
      class="col-md-3 col-12"
    >
      <div
        class="card shadow-sm p-3 mb-4 rounded"
      >
        <div
          class="total-card p-2"
        >
          <div
            class="icon"
            style="background-color: rgb(228, 253, 205);"
          >
            <img
              height="40"
              src="income.png"
              width="40"
            />
          </div>
          <div
            class="labels"
          >
            <h4>
              R 
              NaN
            </h4>
            <p>
              Total Income Before Tax
            </p>
          </div>
        </div>
      </div>
      <div
        class="card shadow-sm p-3 mb-4 rounded"
      >
        <div
          class="total-card p-2"
        >
          <div
            class="icon"
            style="background-color: rgb(255, 238, 210);"
          >
            <img
              height="40"
              src="tax.png"
              width="40"
            />
          </div>
          <div
            class="labels"
          >
            <h4>
              R 
              NaN
            </h4>
            <p>
              Total Income After Tax
            </p>
          </div>
        </div>
      </div>
      <div
        class="card shadow-sm p-3 mb-4 rounded"
      >
        <div
          class="total-card p-2"
        >
          <div
            class="icon"
            style="background-color: rgb(255, 226, 211);"
          >
            <img
              height="40"
              src="expense.png"
              width="40"
            />
          </div>
          <div
            class="labels"
          >
            <h4>
              R 
              NaN
            </h4>
            <p>
              Total Expenses
            </p>
          </div>
        </div>
      </div>
      <div
        class="card shadow-sm p-3 mb-4 rounded"
      >
        <div
          class="total-card p-2"
        >
          <div
            class="icon"
            style="background-color: rgb(221, 221, 247);"
          >
            <img
              height="40"
              src="savings.png"
              width="40"
            />
          </div>
          <div
            class="labels"
          >
            <h4>
              R 
              NaN
            </h4>
            <p>
              Total Savings
            </p>
          </div>
        </div>
      </div>
      <div
        class="card shadow-sm p-3 mb-4 rounded last-total-card"
      >
        <div
          class="final-total p-2"
        >
          <div
            class="icon"
            style="background-color: rgb(207, 242, 109);"
          >
            <img
              height="40"
              src="wallet.png"
              width="40"
            />
          </div>
          <h2>
            R 
            NaN
          </h2>
          <p>
            Total Left To Spend Each Month
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
